File: 1343_numberOfSubarraysOfSizeKAndAverageGreaterThanOrEqualToThreshold/main.cpp
#include <vector>

class Solution {
public:
    int numOfSubarrays(std::vector<int>& arr, int k, int threshold) {
        if(arr.size() < k)
            return -1;
        
        int res = 0;
        threshold = threshold * k;
        int total = 0;
        for(int i = 0; i < k - 1; ++i)
            total = total + arr[i];
        
        for(int r = k-1; r < arr.size(); ++r) {
            total = total + arr[r];
            if(total >= threshold)
                ++res;
            total = total - arr[r-k+1];
        }
        return res;
    }
};

File: 1343_numberOfSubarraysOfSizeKAndAverageGreaterThanOrEqualToThreshold/main.py
from typing import List

class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        if k > len(arr):
            return -1
        res = 0
        threshold *= k
        total = sum(arr[0:k-1])
        for r in range(k-1, len(arr)):
            total += arr[r]
            if total >= threshold:
                res += 1
            total -= arr[r-k+1]
        return res
        

